### PR TITLE
Playground UI improvements and fixes

### LINKED ISSUE #38: Hide Input box from playground until it is asked for.
input box is visible all the time even if the code doesn't have an input  - best if it isn't there in that case

### PR DESCRIPTION
**Pull request by abhishekab**
- Cache code written in shared prefs
- UI improvements

Closes #38, #39. #40

### CODE REVIEW

File: playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt [hunk 1]
```diff
@@ -96,6 +122,24 @@ class PlaygroundActivity : AppCompatActivity() {
             tvOutput.text = "" // clear output window
             viewModel.start(etCode.text.toString())
         }
+
+        etCode.addTextChangedListener( object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+                // cache text to shared Pref
+                sharedPreferences.edit().putString(KEY_PREF_CODE_BACKUP, etCode.text.toString()).apply()
```
Comment (suhaibroomy) [inline playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt:129]: This is should be delegated via viewModel. We can create sharedPrefsProvider class and inject it via koin into viewModel to keep viewModel clean from Android Context class

File: playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt [hunk 2]
```diff
@@ -70,19 +75,40 @@ class PlaygroundActivity : AppCompatActivity() {
...
+            return
+        }
+        super.onBackPressed()
+    }
+
     private fun createCode() {
         etCode = findViewById(R.id.etCode)
 
+        // Restore code from shared Prefs
+        etCode.setText(sharedPreferences.getString(KEY_PREF_CODE_BACKUP,""))
```
Comment (suhaibroomy) [inline playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt:110]: same here, should be delegated via viewModel

Comment (abhishekab) [reply playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt]: Done

File: playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt (hunk 1 above)
Comment (abhishekab) [reply playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt]: Done

**Comment by abhishekab** [comment]
Updated PR

**Pull request merged**